newIDE/app/src/ResourcesList/ResourcePreview/Resource3DPreviewContext.js

```diff
@@ -0,0 +1,268 @@
+// @flow
+import * as React from 'react';
+// Import the worker (will be handled by worker-loader)
+import Resource3DPreviewWorker from './Resource3DPreview.worker';
+
+type WorkerInitMessage = {|
+  type: 'INIT',
+|};
+
+type WorkerRenderModelMessage = {|
+  type: 'RENDER_MODEL',
+  resourceUrl: string,
+|};
+
+type WorkerOutInitMessage = {|
+  type: 'INIT',
+  success: boolean,
+|};
+
+type WorkerOutRenderCompleteMessage = {|
+  type: 'RENDER_COMPLETE',
+  resourceUrl: string,
+  screenshot: string,
+|};
+
+type WorkerOutRenderErrorMessage = {|
+  type: 'RENDER_ERROR',
+  resourceUrl: string,
+  error: string,
+|};
+
+type WorkerOutMessage =
+  | WorkerOutInitMessage
+  | WorkerOutRenderCompleteMessage
+  | WorkerOutRenderErrorMessage;
+
+export type Resource3DPreviewState = {|
+  getResourcePreview: (resourceUrl: string) => Promise<?string>,
+|};
+
+const initialResource3DPreviewState = {
+  getResourcePreview: async (_resourceUrl: string) => null,
+};
+
+const Resource3DPreviewContext = React.createContext<Resource3DPreviewState>(
+  initialResource3DPreviewState
+);
+
+export default Resource3DPreviewContext;
+
+// Message types matching the worker's constants
+const MESSAGE_TYPES = {
+  RENDER_MODEL: 'RENDER_MODEL',
+  RENDER_COMPLETE: 'RENDER_COMPLETE',
+  RENDER_ERROR: 'RENDER_ERROR',
+  INIT: 'INIT',
+};
+
+// Worker manager that handles initialization and communication
+class Resource3DPreviewWorkerManager {
+  worker: Worker;
+  isInitialized: boolean = false;
+  pendingPromises: Map<
+    string,
+    { resolve: Function, reject: Function }
```

Comment:
Function is equivalent to "any" in Flow, prefer:
```suggestion
    { resolve: () => void, reject: () => void }
```

(https://flow.org/en/docs/types/functions/#any-function)